src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthRealms.java

```diff
@@ -77,13 +84,70 @@ public void add(
         {
             throw new IllegalStateException("Too many realms");
         }
-        realmsIdsByName.put(realm, 1L << nextRealmBitShift++);
+        realmsIdsByName.put(realm, new OAuthRealm(1L << nextRealmBitShift++));
     }
 
     public long resolve(
-        String realm)
+        String realm,
+        String[] scopes)
     {
-        return realmsIdsByName.getOrDefault(realm, NO_AUTHORIZATION);
+        // If realm doesn't exist, add it to realms
+        if(!realmsIdsByName.containsKey(realm))
+        {
+            add(realm);
+        }
+        final OAuthRealm realmObject = realmsIdsByName.get(realm);
+        if(realmObject == null)
+        {
+            return NO_AUTHORIZATION;
+        }
+        long realmBit = realmObject.realmBit;
```

Comment:
Rename local `realmBit` variable to `authorization` since that is what we are building up to return from the method after OR-ing in the various scope bits.